web_ui/frontend/app/director/components/DirectorCardList.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {ComponentType, FunctionComponent, useMemo, useState, JSX} from "react";
+import {Box, Grid, Pagination, TextField, Typography} from "@mui/material";
+import {DirectorCard, DirectorCardProps} from "./";
+import {Server} from "@/index";
+import {BooleanToggleButton, CardList} from "@/components";
+import useFuse from "@/helpers/useFuse";
+
+function searchObject<T>(o: T, search: string){
+    const objectString = JSON.stringify(o).toLowerCase()
+    return objectString.includes(search.toLowerCase())
+}
+
+interface DirectorCardListProps {
+    data: Partial<DirectorCardProps>[];
+    cardProps: Partial<DirectorCardProps>;
+}
+
+export function DirectorCardList({ data, cardProps }: DirectorCardListProps) {
+
+    const [search, setSearch] = useState<string>("")
+    const [pelicanServer, setPelicanServer] = useState<boolean | undefined>(undefined)
+    const [serverError, setServerError] = useState<boolean | undefined>(undefined)
+    const [serverDowntime, setServerDowntime] = useState<boolean | undefined>(undefined)
+
+    const searchedData = useFuse<Partial<DirectorCardProps>>(data, search)
+
+    const filteredData = useMemo(() => {
+        let filteredData = structuredClone(searchedData)
+        if (pelicanServer != undefined) {
+            filteredData = filteredData.filter((d) => isPelicanServer(d?.server) == pelicanServer)
```

Comment:
```suggestion
            filteredData = filteredData.filter((d) => isPelicanServer(d?.server) && pelicanServer)
```
why not using `&&` for the three filters?

Reply:
The BooleanToggleButton's value is undefined | true | false based on what value the user wants the associated value to be in the data. 

If I used && then if the user wanted to see only a falsy field they wouldn't be able too.